### PR TITLE
Update demo contract

### PR DESCRIPTION
**Pull request by chenzhitong**
close https://github.com/neo-project/neo-devpack-dotnet/issues/351

### CODE REVIEW

**Review by Tommo-L — APPROVED** [review]

File: templates/Template.CSharp/Contract1.cs [hunk 1]
```diff
@@ -11,10 +11,36 @@ namespace $safeprojectname$
     [Features(ContractFeatures.HasStorage)]
     public class Contract1 : SmartContract
     {
+        //Replace it with your own address.
```
Comment (erikzhang) [inline templates/Template.CSharp/Contract1.cs:14]:
```suggestion
        //TODO: Replace it with your own address.
```

File: templates/Template.CSharp/Contract1.cs [hunk 2]
```diff
@@ -11,10 +11,36 @@ namespace $safeprojectname$
...
+        static readonly byte[] Owner = "NiNmXL8FjEUEs1nfX9uHFBNaenxDHJtmuB".ToScriptHash();
+
+        private static bool IsOwner() => Runtime.CheckWitness(Owner);
+
+        // When this contract address is included in the transaction signature,
+        // this method will be triggered as a VerificationTrigger to verify that the signature is correct.
+        // For example, this method needs to be called when withdrawing token from the contract.
+        public static bool Verify() => IsOwner();
+
         public static bool Main()
```
Comment (erikzhang) [inline templates/Template.CSharp/Contract1.cs:24]:
```suggestion
        public static bool MyMethod()
```

Comment (erikzhang) [reply templates/Template.CSharp/Contract1.cs:24]: And we can add some comments to explain that this should be replaced with your own methods.

File: templates/Template.CSharp/Contract1.cs [hunk 3]
```diff
@@ -11,10 +11,36 @@ namespace $safeprojectname$
...
         {
             Storage.Put("Hello", "World");
             return true;
         }
+
+        public static bool Update(byte[] script, string manifest)
+        {
+            if (!IsOwner()) throw new Exception("No authorization.");
+            // Check empty
+            if (script.Length == 0 && manifest.Length == 0) return false;
```
Comment (erikzhang) [inline templates/Template.CSharp/Contract1.cs:34]: No need to check it. `Contract.Update()` will check it for you.

File: templates/Template.CSharp/Contract1.cs [hunk 4]
```diff
@@ -11,10 +11,36 @@ namespace $safeprojectname$
...
+            Contract.Update(script, manifest);
+            return true;
+        }
+
+        public static bool Destroy()
```
Comment (erikzhang) [inline templates/Template.CSharp/Contract1.cs:39]: Return `void`.

**Review by shargon — DISMISSED** [review]

File: templates/Template.CSharp/Contract1.cs [hunk 5]
```diff
@@ -11,10 +11,32 @@ namespace $safeprojectname$
...
+        static readonly byte[] Owner = "NiNmXL8FjEUEs1nfX9uHFBNaenxDHJtmuB".ToScriptHash();
+
+        private static bool IsOwner() => Runtime.CheckWitness(Owner);
+
+        // When this contract address is included in the transaction signature,
+        // this method will be triggered as a VerificationTrigger to verify that the signature is correct.
+        // For example, this method needs to be called when withdrawing token from the contract.
+        public static bool Verify() => IsOwner();
+
+        public static bool MyMethod()
```
Comment (Tommo-L) [inline templates/Template.CSharp/Contract1.cs:24]:
```suggestion
        // TODO: Change the name
        public static bool MyMethod()
```

**Review by erikzhang — APPROVED** [review]

**Review by shargon — APPROVED** [review]

**Pull request merged**